ui/src/dashboards/containers/DashboardPage.js

```diff
@@ -57,13 +58,23 @@ class DashboardPage extends Component {
     this.synchronizer = ::this.synchronizer
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     const {
       params: {dashboardID},
-      dashboardActions: {getDashboardsAsync},
+      dashboardActions: {
+        getDashboardsAsync,
+        updateTempVarValues,
+        putDashboardByID,
+      },
+      source,
     } = this.props
 
-    getDashboardsAsync(dashboardID)
+    const dashboards = await getDashboardsAsync(dashboardID)
```

Comment:
also, while you're at it, this is old code, but `getDashboardsAsync` doesn't accept any arguments, so passing in `dashboardID` here is misleading